default.nix

```diff
@@ -57,12 +63,12 @@ let
   # fetched binaries are the "standard" builds that people test. This should be
   # fast as it mostly fetches Hydra (CI) caches without building much.
   cardano-node = import
-    (pkgs.fetchgit {
+    (builtins.fetchGit {
       url = "https://github.com/input-output-hk/cardano-node";
-      rev = "1.35.3";
-      sha256 = "020fwimsm24yblr1fmnwx240wj8r3x715p89cpjgnnd8axwf32p0";
+      ref = "purity";
+      rev = "56a7c0ba48e363a65bed4b876c7e14005eab878d";
```

Comment:
why rev is not "1.35.3" ?

Reply:
We need https://github.com/input-output-hk/cardano-node/pull/4661 for `shell.nix`